.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,241 @@
+name: FloPy benchmarks
+
+on:
+  schedule:
+    - cron: '0 8 * * *' # run at 8 AM UTC (12 am PST)
+
+jobs:
+  benchmark:
+    name: Benchmarks
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        python-version: [ 3.7, 3.8, 3.9, "3.10" ]
+        exclude:
+          # avoid shutil.copytree infinite recursion bug
+          # https://github.com/python/cpython/pull/17098
+          - python-version: '3.8.0'
+        include:
+          - os: ubuntu-latest
+            path: ~/.cache/pip
+          - os: macos-latest
+            path: ~/Library/Caches/pip
+    defaults:
+      run:
+        shell: bash
+    timeout-minutes: 90
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2.3.4
+
+      - name: Cache Python
+        uses: actions/cache@v3
+        with:
+          path: ${{ matrix.path }}
+          key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('setup.cfg') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python-version }}-pip-
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Get branch name
+        uses: nelonoel/branch-name@v1.0.1
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .
+          pip install ".[test, optional]"
+
+      - name: Cache Modflow executables
+        uses: actions/cache@v3
+        with:
+          path: $HOME/.local/bin
+          key: modflow-exes-${{ matrix.os }}
```

Comment:
How will the cache be updated when there is a new release on the modflow-executables repo?

Here and elsewhere. 

code.json in the zip files has the version numbers. We could add the code.json to the release and maybe that could be hashed so that the latest will always be used.

Reply:
Hashing `code.json` should work, flopy CI would just need to check out the executables repo.

That seems like a nicer solution than invalidating the cache via API per https://github.com/MODFLOW-USGS/executables/pull/7, I'll try it.

Reply:
I was thinking of adding it as a release asset but could add it as a file to the repo. If it is a release asset, it could be pulled using the GitHub API before the MODFLOW executable step in the workflows. Adding it as a file in the repo would allow us to tag the commit and tie the release to the tag. I think we would have to make a new release after adding code.json to the repo to test this option.

Reply:
The release asset option might be simplest. I'll update this PR to look for `code.json` in the executables repo release assets & use its hash in the cache key if it finds one, otherwise just using `modflow-exes-<system>` as above. That should allow caching to work now and if/when we do include `code.json` as a release asset.

Marked as draft again, will re-mark for review when done.